cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/UnreachableEOGPass.kt

```diff
@@ -59,142 +62,143 @@ class UnreachableEOGPass(ctx: TranslationContext) : TranslationUnitPass(ctx) {
      * @param node every node in the TranslationResult
      */
     protected fun handle(node: Node, parent: Node?) {
-        if (node is FunctionDeclaration) {
-            val startState = UnreachabilityState()
-            for (firstEdge in node.nextEOGEdges) {
-                startState.push(firstEdge, ReachabilityLattice(Reachability.REACHABLE))
-            }
-            val finalState = iterateEOG(node.nextEOGEdges, startState, ::transfer) ?: return
+        val startState = UnreachabilityState()
```

Comment:
We should not run this for every node. The SubgraphWalker will iterate the whole graph and we should select some meaningful entry points for the analysis. So far, these were function declarations. It may make sense to use the first EOG starter node in the EOG instead but not all nodes in the graph

Reply:
I changed the pass already to an eog starter pass so this is actually doing exactly that :)

Reply:
Right, my fault